core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/ExprDsl.scala

```diff
@@ -10,17 +10,19 @@ import com.rallyhealth.vapors.core.math.{Addition, Negative, Subtraction}
 import com.rallyhealth.vapors.factfilter.data._
 import com.rallyhealth.vapors.factfilter.evaluator.InterpretExprAsFunction
 
+import scala.collection.immutable.SortedSet
+
 object ExprDsl extends ExprBuilderSyntax {
 
-  type CondExpr[F[_], V, P] = Expr[F, V, Boolean, P]
+  final type CondExpr[F[_], V, P] = Expr[F, V, Boolean, P]
```

Comment:
Does this being an object not already make these final? Should the rest of these be final? It's a shame that final is not the default in Scala.

Reply:
I'm not sure... I just put `final` everywhere because, as you say, it should be the default. I believe Scala 3 uses a new `open` keyword for making a class open for extension, and in Scala >3.1, it will be required to extend a class and all non-open classes will be automatically made `final`.